### ISSUE TITLE
Wrong absolute paths to gifs used in component

### ISSUE THREAD

**Issue by nadzic** [teambit/bit#1579]
## Expected Behavior

External asset of component (gif) should be positioned on the right folder.

## Actual Behavior

External assest in wrong folder or path to asset (gif) is wrong.

## Steps to Reproduce the Problem

I developed following component: https://bit.dev/nadzic/test/loader/~code (exported it with omg_loader.gif) and tried to reuse in another project. So when I reuse it, I get following error:

<img width="1127" alt="Screenshot 2019-04-29 at 14 26 33" src="https://user-images.githubusercontent.com/6192073/56896258-87a13c00-6a8b-11e9-86f4-3caa61214d3e.png">

Because the omg_loader.gif is not in wanted folder. The folder structure looks like this:

<img width="618" alt="Screenshot 2019-04-29 at 14 34 13" src="https://user-images.githubusercontent.com/6192073/56896421-fe3e3980-6a8b-11e9-8663-5fb3719e8215.png">
<img width="992" alt="Screenshot 2019-04-29 at 14 34 41" src="https://user-images.githubusercontent.com/6192073/56896422-fe3e3980-6a8b-11e9-8c55-29027294a211.png">

## Specifications

- Bit version: 14.0.6
- Node version: v8.6.0
- npm / yarn version: 6.4.1
- Platform: Mac os X
- Bit compiler (include version): bit.envs/compilers/react@1.0.1
- Bit tester (include version): bit.envs/testers/jest@0.0.35

Is there only an option to use relative links for resources or is there any option that we keep absolute urls? For both apps I tried to set app name ldmaapp.

**Comment by davidfirst** [comment]
It's a bug. 
It happens when these three conditions met: 
1. using binary files (.gif in this case). 
2. using custom-resolve-modules. (`require('ldmaapp/assets/gif/omg_loader.gif')`) in this case).
3. requiring the file within the same component and not as a dependency.

As a workaround, you can change #2 or #3 and it should work. In other words, try one of the following:
1. require the `.gif` file with relative syntax. (something like `require('../../assets/gif/omg_loader.gif')`
2. export the gif file as a separate component and require it from the loader component.

**Comment by nadzic** [comment]
@davidfirst @GiladShoham Are you sure it is fixed?

I upgraded bit to 14.1.0, re-exported the component and reinstalled node_modules in the project where I want to use the component and still getting the following:

<img width="636" alt="Screenshot 2019-05-02 at 15 09 03" src="https://user-images.githubusercontent.com/6192073/57077393-3f348900-6cec-11e9-9df7-2d2937fa7e06.png">

<img width="636" alt="Screenshot 2019-05-02 at 15 09 33" src="https://user-images.githubusercontent.com/6192073/57077426-51162c00-6cec-11e9-9903-c0d5deccd267.png">

Here is the link to the component:

https://bit.dev/nadzic/test/loader/~code

**Comment by nadzic** [comment]
Maybe you can try to install it by yourself

**Comment by GiladShoham** [comment]
@nadzic Thanks for the update, we will check it again and post an update soon

**Comment by nadzic** [comment]
The issue is, in file node_modules/nadzic.test.loader/dist/src/components/common/Loader.js is always requring in the following way: source:require('ldmaapp/assets/gif/omg_loader.gif') - even if I set relative paths in component, as in example: https://bit.dev/nadzic/test/loader/~code

**Comment by davidfirst** [comment]
@nadzic , the fix above was to get the module path (`require('ldmaapp/assets/gif/omg_loader.gif')`) working instead of the relative.
The relative syntax should work regardless of Bit because both files are in the same component, so Bit doesn't generate any link here.
I tried to install your component using NPM and in the dist I see the following:
`source:require('../../../assets/gif/omg_loader.gif')`, which looks good to me.
Where do you get the error above? maybe you can paste here a snippet I could run to get that error.

**Comment by nadzic** [comment]
@davidfirst Hi!

1. I tried the option that omg_loader (gif) was its own component and it did not work. Never mind - leave that option for now.
2. I tried to leave the option with absolute path, I also upgraded bit to 14.1.0 before the export and I was getting the same issue.
3. As 2. version did not work for me, I tired with relative path and was getting in file node_modules/nadzic.test.loader/dist/src/components/common/Loader.js after remove node_modules and reinstall node_modules. So, maybe I am assuming is problem with npm cache that I did not get the same component content as you did?

I will try to reinstall node_modules, clean cache and will report you again here.
